File: prime_number/Python/sieve.py
import sys
import array
import math
isprime=array.array('i',(0 for i in range(0,1001)))
def sieve():
    for i in range(2,math.sqrt(1000)):
        for j in range(2,sys.maxsize**10):
            if(i*j >1000):
                break
            isprime[i*j]=1
sieve()
n=int(input())
for i in range(2,n+1):
    if(isprime[i] is 0):
        print(i,end=" ")

File: sieve_of_eratosthenes/sieve.js
const rl = require('readline');

function sieve(n) {
  const keys = new Array(n).fill(true)

  for (let i = 2; i < Math.sqrt(n); i++) {
    if (keys[i]) {
      for (let j = Math.pow(i, 2); j < n; j += i) {
        keys[j] = false
      }
    }
  }

  return keys.reduce((acc, val, i) => [
    ...acc,
    ...(val ? [i] : [])
  ], []).slice(2);
}

// Quick n' dirty input
const input = rl.createInterface(process.stdin, process.stdout, null);
input.question("Get the primes up to: ", function(answer) {
  const primes = sieve(parseInt(answer));
  console.log(primes);
  input.close();
  process.stdin.destroy();
});